tmt/_version.py

```diff
@@ -0,0 +1,3 @@
+# file generated by setuptools_scm
+# don't change, don't track in version control
+__version__ = version = 'running from source'
```

Comment:
Is this expected to be included? Even with the `.gitignore` pattern added, changes to this file are tracked:

```
> git status
On branch hatch
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tmt/_version.py
```

Reply:
We're doing this in tmt/init.py: `from tmt._version import version as __version__`.
Without the file being present, mypy, tmt lint would fail. 

How about `git update-index --skip-worktree tmt/_version.py`? Could you try if it's still being tracked?

Reply:
Yes, this seems to hide the updated file from the user. One more thought: What about restoring the file content after the version is no more needed (e.g. package built)?

Reply:
You mean as a part of the `clean` target in Makefile? What would be the best approach? Back it up first?

Reply:
Hm, maybe `clean`. Before we had some places where temporarily files were backed up, like [this one](https://github.com/teemtee/tmt/blob/main/Makefile#L76-L79). So maybe in a similar way?

Reply:
> How about `git update-index --skip-worktree tmt/_version.py`? Could you try if it's still being tracked?

This does not seem right. It just hides the modified file but it's still waiting there until for example:

```
> git checkout main
error: Your local changes to the following files would be overwritten by checkout:
	tmt/_version.py
Please commit your changes or stash them before you switch branches.
Aborting
```

So, I guess, this is not the way.

Reply:
Also discovered this. I'm backing/restoring the in the `rpm` target. Seems to work fine.
What to do with _version.py? Put it back to .gitignore? Remove skip-worktree?

Reply:
A bit trickier. backing/restoring the _version.py file is easy as it's only needed for `hatch build`. The specfile needs to be modified in multiple targets.

I think we could use env var, or something like `--define "version %(hatch version)"`, but I'm not confident I can find the correct solution. Any suggestions?

Reply:
I think it would be best to just not create _version.py and use importlib - something like:
```from importlib.metadata import version
version(__name__)
```
as it seems to be the 'modern' approach and could prevent some headaches. 
It could also be done only where needed instead of _init. 

Not sure if you want this as part of this PR though @psss ?

As for non-packit local rpm builds which needs .spec edit to match version - are there any rpmbuild experts to help with the best way to do this using macros/variables/env variables in Makefile? (Something like the --define example in previous comment)

Reply:
Sounds tempting. Let's polish this in a separate pull request.